### PR TITLE
fix(pkg/client/clienttest): enforce mock client to interface

### PR DESCRIPTION
**Pull request by arriqaaq**
These changes are required for https://github.com/codenotary/immugw to work, as it depends on mock clients from client test

- add interface assertion for immuclient
- add interface assertion for immuServiceClient
- add interface assertion for tokenservice

References: https://github.com/codenotary/immudb-issues/issues/76

### CODE REVIEW

File: pkg/client/clienttest/immuServiceClient_mock.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ import (
 	"google.golang.org/grpc"
 )
 
+var (
+	_ schema.ImmuServiceClient = &ImmuServiceClientMock{}
```
Comment (jeroiraz) [inline pkg/client/clienttest/immuServiceClient_mock.go:29]:
```suggestion
	_ schema.ImmuServiceClient = (*ImmuServiceClientMock)(nil)
```

File: pkg/client/clienttest/immuclient_mock.go [hunk 2]
```diff
@@ -27,6 +27,10 @@ import (
 	"google.golang.org/grpc"
 )
 
+var (
+	_ client.ImmuClient = &ImmuClientMock{}
```
Comment (jeroiraz) [inline pkg/client/clienttest/immuclient_mock.go:31]:
```suggestion
	_ client.ImmuClient = (*ImmuClientMock)(nil)
```

File: pkg/client/clienttest/token_service_mock.go [hunk 3]
```diff
@@ -21,6 +21,10 @@ import (
 	"github.com/codenotary/immudb/pkg/client/tokenservice"
 )
 
+var (
+	_ tokenservice.TokenService = &TokenServiceMock{}
```
Comment (jeroiraz) [inline pkg/client/clienttest/token_service_mock.go:25]:
```suggestion
	_ tokenservice.TokenService = (*TokenServiceMock)(nil)
```

File: pkg/client/clienttest/homedir_mock.go [hunk 4]
```diff
@@ -21,6 +21,10 @@ import (
 	"github.com/codenotary/immudb/pkg/client/homedir"
 )
 
+var (
```
[the remark this answers is not in the document]
Comment (arriqaaq) [reply pkg/client/clienttest/homedir_mock.go:24]: done

File: pkg/client/clienttest/token_service_mock.go (hunk 3 above)
Comment (arriqaaq) [reply pkg/client/clienttest/token_service_mock.go]: done

File: pkg/client/clienttest/immuServiceClient_mock.go (hunk 1 above)
Comment (arriqaaq) [reply pkg/client/clienttest/immuServiceClient_mock.go]: done

**Review by jeroiraz — APPROVED** [review]

**Pull request merged**